pysat/_instrument.py

```diff
@@ -1973,11 +1963,36 @@ def date(self, new_date):
         # Set the date property, see property docstring for details
         self._date = utils.time.filter_datetime_input(new_date)
 
+    @property
+    def empty(self):
+        """Boolean flag reflecting lack of data, True if there is no data."""
+        return self._empty()
+
     @property
     def index(self):
         """Time index of the loaded data."""
         return self._index()
 
+    @property
+    def pandas_format(self):
+        """Boolean flag for pandas data support."""
+        return self._pandas_format
+
+    @pandas_format.setter
```

Comment:
Recommend raising an error if attempting to update this after data has been loaded.

Reply:
That is a good point. Though something like wavelet processing could trigger a type change from pandas to xarray to store the newly calculated data. We may want to leave the ability to change an existing Instrument object data type but I agree it does need something.

Reply:
> Though something like wavelet processing could trigger a type change from pandas to xarray to store the newly calculated data

This is certainly something we should consider!  But I think this is an enhancement that will require other (extensive?) changes as well.  I would recommend making that an issue, and making it impossible to do at the moment (while referencing that the error should change once the issue is addressed).

Reply:
I just came back here to remind myself not to go adding new features! 🙈 Fortunately you were on the ball ahead of me.

Good news though, doesn't have to be an extra feature. If a user wants to convert data types via custom functions they can remove the existing data, thus making the instrument empty, change the type, and then add the new one.

My first thought for the check in the pandas_format setter is an inst.empty check. Offhand I don't remember if we have a flag that indicates if data has ever been loaded.